syntax = "proto3";

package syft.core.node.common.action;

import "proto/core/common/common_object.proto";
import "proto/core/store/store_object.proto";
import "proto/core/io/address.proto";

message GetObjectAction {
  syft.core.common.UID id_at_location = 1;
  syft.core.common.UID msg_id = 2;
  syft.core.io.Address address = 3;
  syft.core.io.Address reply_to = 4;
  bool delete_obj = 5;
}

message GetObjectResponseMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.store.StorableObject obj = 2;
  syft.core.io.Address address = 3;
}